modern/gcc/specs

```diff
@@ -0,0 +1,145 @@
+
+*lib:
+%{!mandroid|tno-android-ld:%{pthread:-lpthread} %{shared:-lc}    %{!shared:%{profile:-lc_p}%{!profile:-lc}};:%{shared:-lc}    %{!shared:%{profile:-lc_p}%{!profile:-lc}} %{!static: -ldl}}
+
+*link_gomp:
+
+
+*libgcc:
+%{static|static-libgcc|static-pie:-lgcc -lgcc_eh}%{!static:%{!static-libgcc:%{!static-pie:%{!shared-libgcc:-lgcc --push-state --as-needed -lgcc_s --pop-state}%{shared-libgcc:-lgcc_s%{!shared: -lgcc}}}}}
+
+*startfile:
+%{!mandroid|tno-android-ld:%{shared:;      pg|p|profile:%{static-pie:grcrt1.o%s;:gcrt1.o%s};      static:crt1.o%s;      static-pie:rcrt1.o%s;      pie:Scrt1.o%s;      :crt1.o%s} crti.o%s    %{static:crtbeginT.o%s;      shared|static-pie|pie:crtbeginS.o%s;      :crtbegin.o%s}    %{fvtable-verify=none:%s;      fvtable-verify=preinit:vtv_start_preinit.o%s;      fvtable-verify=std:vtv_start.o%s} ;:%{shared: crtbegin_so%O%s;:  %{static: crtbegin_static%O%s;: crtbegin_dynamic%O%s}}}
+
+*cross_compile:
+0
+
+*version:
+11.4.0
+
+*multilib:
+. !m64 !m32;.:../lib64:x86_64-linux-gnu m64 !m32;.:../lib32:i386-linux-gnu !m64 m32;
+
+*multilib_defaults:
+m64
+
+*multilib_extra:
+
+
+*multilib_matches:
+m64 m64;m32 m32;
+
+*multilib_exclusions:
+
+
+*multilib_options:
+m64/m32
+
+*multilib_reuse:
+
+
+*linker:
+collect2
+
+*linker_plugin_file:
+
+
+*lto_wrapper:
+
+
+*lto_gcc:
+
+
+*post_link:
+
+
+*link_libgcc:
+%D
+
+*md_exec_prefix:
+
+
+*md_startfile_prefix:
+
+
+*md_startfile_prefix_1:
+
+
+*startfile_prefix_spec:
+
+
+*sysroot_spec:
+--sysroot=%R
+
+*sysroot_suffix_spec:
+
+
+*sysroot_hdrs_suffix_spec:
+
+
+*self_spec:
+
+
+*cc1_cpu:
+%{march=native:%>march=native %:local_cpu_detect(arch %{!m32:64;:32})   %{!mtune=*:%>mtune=native %:local_cpu_detect(tune %{!m32:64;:32})}} %{mtune=native:%>mtune=native %:local_cpu_detect(tune %{!m32:64;:32})}
+
+*link_command:
```

Comment:
From https://gcc.gnu.org/onlinedocs/gcc/Spec-Files.html I get the idea that we could be defining only this, leaving the rest to use the built-in ones?

Reply:
The changes are un `link`, `libgcc` and `link_command`, the latter being the biggest portion of the file. If this file 100% overrides the spec, I would probably leave it as is: it can be diffed with the built-in spec so we know exactly what the changes are, we are only doing it for gcc11, and we have plans in the future to not need this moving forward.

I wouldn't necessarily want to experiment with which specs are defined first by gcc or not if this approach works